### ISSUE TITLE
Incorrect record use_Stanley_ML when USE_BODNER23 is true

### ISSUE THREAD

**Issue by gustavo-marques** [NCAR/MOM6#328]
I think parameter `CS%use_Stanley_M`L is being recorded incorrectly when `USE_BODNER23` is true; see [here](https://github.com/NCAR/MOM6/blob/acadc6eca5808b302c5d5dbb2bece8ea05aaf316/src/parameterizations/lateral/MOM_mixed_layer_restrat.F90#L1638-L1640).

First, `use_Stanley_ML` is not implemented in subroutine `mixedlayer_restrat_Bodner`, so making it available with this option does not make sense. 

Second, were `use_Stanley_ML` a valid option for `mixedlayer_restrat_Bodner`, the `get_param` should use `USE_STANLEY_ML` instead of `USE_STANLEY_TVAR`.

I suggest we remove [this](https://github.com/NCAR/MOM6/blob/acadc6eca5808b302c5d5dbb2bece8ea05aaf316/src/parameterizations/lateral/MOM_mixed_layer_restrat.F90#L1638-L1640). call to avoid confusion. 

This can be added to https://github.com/NCAR/MOM6/pull/327

**Comment by iangrooms** [comment]
Agreed. I will put it on my list to add the Stanley density correction to the Bodner ML code.

**Comment by iangrooms** [comment]
#359 addresses this while also cleaning up the Stanley parameterization logic and adding some documentation.

**Comment by iangrooms** [comment]
This can be closed. Fixed in #359.
